pkg/kubelet/status_manager.go

```diff
@@ -151,3 +168,14 @@ func (s *statusManager) syncBatch() error {
 	s.DeletePodStatus(podFullName)
 	return fmt.Errorf("error updating status for pod %q: %v", pod.Name, err)
 }
+
+// notRunning returns true if every status is terminated or waiting, or the status list
+// is empty.
+func notRunning(statuses []api.ContainerStatus) bool {
+	for _, status := range statuses {
+		if status.State.Terminated == nil && status.State.Waiting == nil {
```

Comment:
If someone adds a condition that is a subset of Running this would break.  I can make that change if you think it's safer.

----- Original Message -----
> > @@ -151,3 +168,14 @@ func (s *statusManager) syncBatch() error {
> >  	s.DeletePodStatus(podFullName)
> >  	return fmt.Errorf("error updating status for pod %q: %v", pod.Name, err)
> >  }
> > +
> > +// notRunning returns true if every status is terminated or waiting, or
> > the status list
> > +// is empty.
> > +func notRunning(statuses []api.ContainerStatus) bool {
> > +	for _, status := range statuses {
> > +		if status.State.Terminated == nil && status.State.Waiting == nil {
> 
> How about changing the function to something like "stillRunning", and replace
> this with `if status.State.Running != nil { return true}` ?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/9165/files#r31835173